lib/ansible/parsing/yaml/dumper.py

```diff
@@ -23,16 +23,24 @@
 
 from ansible.module_utils.six import PY3
 from ansible.parsing.yaml.objects import AnsibleUnicode, AnsibleSequence, AnsibleMapping, AnsibleVaultEncryptedUnicode
+from ansible.parsing.yaml.utils import convert_yaml_objects_to_native
 from ansible.utils.unsafe_proxy import AnsibleUnsafeText, AnsibleUnsafeBytes
+from ansible.utils.yaml import HAS_LIBYAML, SafeDumper
 from ansible.vars.hostvars import HostVars, HostVarsVars
 
 
-class AnsibleDumper(yaml.SafeDumper):
+class AnsibleDumper(SafeDumper):
     '''
     A simple stub class that allows us to add representers
     for our overridden object types.
     '''
-    pass
+    def __init__(self, *args, **kwargs):
+        super(AnsibleDumper, self).__init__(*args, **kwargs)
+
+    def represent(self, data):
+        if HAS_LIBYAML:
+            data = convert_yaml_objects_to_native(data)
```

Comment:
I don't especially love this, but since the C extension doesn't allow for representing str subclasses, there's really no way to handle this. And with the number of libraries I have seen that monkeypatch pyyaml, doing `yaml.SafeDumper = yaml.CSafeDumper`, I think ultimately this is the safest thing to do.

This effectively brings YAML encoding into alignment with what we've had to do with JSON and TOML.